include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -781,6 +781,13 @@ struct SessionImpl : ConstSessionImpl<T> {
 
   void Run(const RunOptions& run_options, const IoBinding&);  ///< Wraps OrtApi::RunWithBinding
 
+  /** \brief Run the model in a separate thread.
+   * Callback will be invoked on run completion, with output values as arguments,
+   * on error, a status could be returned.
+   */
+  void RunAsync(const RunOptions& run_options, const char* const* input_names, const Value* input_values, size_t input_count,
+                const char* const* output_names, Value* output_values, size_t output_count, RunAsyncCallbackFn callback, void* user_data);
+
```

Comment:
Please document args including if the user owns the outputs. The user shouldn't have to go find the underlying C API function called to find the doco.

Reply:
Done.